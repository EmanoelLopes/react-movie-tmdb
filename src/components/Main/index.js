import React, { useEffect } from 'react';
import { node } from 'prop-types';
function Main ({ children = {} }) {
  useEffect(() => {
    let unmounted = true;

    if (!unmounted) {
      window.scrollTo(0, 0);
    }

    return () => unmounted = false;
  }, []);

  return (
    <main className="rm-main">
      {children}
    </main>
  );
}
 
Main.propTypes = {
  children: node,
};

export default Main;
