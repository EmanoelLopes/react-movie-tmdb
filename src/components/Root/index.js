import React from 'react';
import { node } from 'prop-types';
import { Provider } from 'react-redux';
import store from 'store';

const Root = ({ children }) => (
  <Provider store={store}>
    {children}
  </Provider>
);

Root.propTypes = {
  children: node,
};

Root.defaultProps = {
  children: {},
};

export default Root;
