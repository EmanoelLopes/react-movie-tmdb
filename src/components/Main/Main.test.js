import React from 'react';
import { mount } from 'enzyme';
import Root from 'components/Root';
import Main from 'components/Main';

let wrapped;
let defaultProps = {
  children: `<div>Main Component</div>`,
};

beforeEach(() => {
  wrapped = mount(
    <Root>
      <Main {...defaultProps} />
    </Root>
  );
});

afterEach(() => {
  wrapped.unmount();
});

describe('<Main />', () => {
  it('should match snapshot', () => {
    expect(wrapped).toMatchSnapshot();
  });
});
