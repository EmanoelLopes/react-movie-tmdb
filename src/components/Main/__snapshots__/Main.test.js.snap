// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshot 1`] = `
<Root>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Main>
      <main
        className="rm-main"
      >
        &lt;div&gt;Main Component&lt;/div&gt;
      </main>
    </Main>
  </Provider>
</Root>
`;
